### PR TITLE
Constructor Optimizations

### PR DESCRIPTION
**Pull request by bengarney**
Added allocation benchmark. 10k allocations went from 335ms to 200ms avg time on 2.6ghz MacBook Pro (Retina, 13-inch, Mid 2014).
Constructors now call their own supers.
Instance initializer blocks are also now part of constructor, but are also kept separately for use when wrapping natives.
Tweaked script instance memory allocation to achieve better performance (on OSX).
Type now caches many frequently called queries to avoid massive data structure traversals during object construction.
Console::Log now avoids memory allocation.
Tested with JIT and Classic Lua.

### CODE REVIEW

**Comment by bengarney** [comment]
This fails the native managed wrapper unit test, so I need to fix that before it lands.

File: docs/examples/ScriptBenchmark/src/BenchmarkAllocation.ls [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        }
+
+        public var x = 1.0;
+        public var y = 2.0;
+        public var z = 3.0;
+        public var w = 0.0;
+    }
+
+    /*
+     * Benchmark for raw field access
```
Comment (bengarney) [inline docs/examples/ScriptBenchmark/src/BenchmarkAllocation.ls]: This is inaccurate.

File: loom/graphics/CMakeLists.txt [hunk 2]
```diff
@@ -60,6 +57,8 @@ if (FLASH)
 
 endif()
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x" )
```
Comment (themuflon) [inline loom/graphics/CMakeLists.txt]: https://github.com/LoomSDK/LoomSDK/blob/JG-ShaderObjects/CMakeLists.txt#L75

I have already fixed this in my shader objects branch. Also, only clang/gcc accept this parameter. Msvc generates a warning, it's enabled by default anyway.

File: docs/examples/ScriptBenchmark/src/BenchmarkAllocation.ls (hunk 1 above)
Comment (SmilyOrg) [inline docs/examples/ScriptBenchmark/src/BenchmarkAllocation.ls]: Benchmark for inherited object instantiation?

File: loom/engine/loom2d/l2dQuad.cpp [hunk 3]
```diff
@@ -129,10 +129,7 @@ void Quad::render(lua_State *L)
 
     if (renderState.isClipping()) GFX::Graphics::setClipRect((int)renderState.clipRect.x, (int)renderState.clipRect.y, (int)renderState.clipRect.width, (int)renderState.clipRect.height);
 
-    GFX::VertexPosColorTex *v   = GFX::QuadRenderer::getQuadVertices(nativeTextureID, 
-                                                                        4, 
-                                                                        tinted || renderState.alpha != 1.f,
-                                                                        blendSrc, blendDst);
+    GFX::VertexPosColorTex *v = GFX::QuadRenderer::getQuadVertexMemory(4, nativeTextureID, blendSrc, blendDst);
```
Comment (SmilyOrg) [inline loom/engine/loom2d/l2dQuad.cpp]: Should these quad/quadbatch changes be in this branch?

File: loom/script/compiler/lsJitTypeCompiler.cpp [hunk 4]
```diff
@@ -34,6 +35,185 @@ extern "C" {
...
+        return OPR_OR;
+    }
+
+    if (t == &tok->OPERATOR_CONCAT)
+    {
+        return OPR_CONCAT;
+    }
+
+    /*case '^': return OPR_POW;
+     * case TK_CONCAT: return OPR_CONCAT;*/
```
Comment (SmilyOrg) [inline loom/script/compiler/lsJitTypeCompiler.cpp]: Obsolete code or missed case?

File: loom/script/compiler/lsJitTypeCompiler.cpp [hunk 5]
```diff
@@ -226,8 +387,90 @@ void JitTypeCompiler::generateConstructor(FunctionLiteral *function,
...
+
+    if(constructor->getDeclaringType() == NULL
+       || constructor->getDeclaringType()->getBaseType() == NULL
+       || constructor->getDeclaringType()->getBaseType()->isInterface()
+       || (constructor->getDeclaringType()->getBaseType()->getConstructor()->isNative()
+           && constructor->getDeclaringType()->getBaseType()->getFullName() != "system.BaseDelegate")
+       || constructor->getDeclaringType()->getBaseType()->getFullName() == "system.Object")
+        skipSuper = true;
+
+    //printf("Considering super for %s\n", constructor->getDeclaringType()->getFullName().c_str());
```
Comment (SmilyOrg) [inline loom/script/compiler/lsJitTypeCompiler.cpp]: Dead code?

File: loom/script/compiler/lsJitTypeCompiler.cpp [hunk 6]
```diff
@@ -226,8 +387,90 @@ void JitTypeCompiler::generateConstructor(FunctionLiteral *function,
...
+    // it at runtime which is a performance overhead.
+    if(function->isConstructor
+       && function->hasSuperCall == false
+       && function->isNative == false
+       && skipSuper == false
+       && constructor->getDeclaringType()->isPrimitive() == false)
+    {
+        //printf("Generating super call to %s in constructor of %s\n",
+        //       constructor->getDeclaringType()->getBaseType() ? constructor->getDeclaringType()->getBaseType()->getFullName().c_str() : "(none)",
+        //       constructor->getDeclaringType()->getFullName().c_str());
```
Comment (SmilyOrg) [inline loom/script/compiler/lsJitTypeCompiler.cpp]: Could put it under a debug switch if useful?

File: loom/script/compiler/lsJitTypeCompiler.cpp [hunk 7]
```diff
@@ -1031,7 +1151,7 @@ void JitTypeCompiler::block(CodeState *cs, Statement *fstat)
     BC::lineNumber = fstat->lineNumber;
     fstat->visitStatement(this);
 
-    //assert(bl.breaklist == NO_JUMP);
+    lmAssert(bl.breaklist == NO_JMP, "Internal Compiler Error");
```
Comment (SmilyOrg) [inline loom/script/compiler/lsJitTypeCompiler.cpp]: Are `NO_JUMP` and `NO_JMP` the same thing?

File: loom/script/compiler/lsJitTypeCompiler.cpp [hunk 8]
```diff
@@ -1158,85 +1235,13 @@ void JitTypeCompiler::generatePropertySet(ExpDesc *call, Expression *value,
...
-    return OPR_NOBITOPR;
-}
-
-
 Expression *JitTypeCompiler::visit(AssignmentOperatorExpression *expression)
 {
     Tokens *tok = Tokens::getSingletonPtr();
     BinOpr op   = getassignmentopr(expression->type);
 
+    //lmAssert(op != OPR_NOBINOPR, "Unknown bin op on AssignentOperatorExpression");
```
Comment (SmilyOrg) [inline loom/script/compiler/lsJitTypeCompiler.cpp]: Dead code?

File: loom/script/compiler/lsJitTypeCompiler.cpp [hunk 9]
```diff
@@ -1654,327 +1547,92 @@ Expression *JitTypeCompiler::visit(BinaryOperatorExpression *expression)
...
-    if (!expressions && vararg)
-    {
-        lmAssert(vararg->position == 0, "Bad position on variable args");
-        createVarArg(e, NULL, vararg->position);
-        return 1;
-    }
+    flist = encodeCondition(expr);
 
-    int count = 0;
+    /*true statement*/
```
Comment (SmilyOrg) [inline loom/script/compiler/lsJitTypeCompiler.cpp]: What does this mean?

File: loom/script/compiler/lsJitTypeCompiler.cpp [hunk 10]
```diff
@@ -2389,7 +1803,7 @@ Expression *JitTypeCompiler::visit(FunctionLiteral *literal)
     inLocalFunction++;
 
     char funcname[256];
-    sprintf(funcname, "__ls_localfunction%i", functioncount++);
+    snprintf(funcname, 250, "__ls_localfunction%i", functioncount++);
```
Comment (SmilyOrg) [inline loom/script/compiler/lsJitTypeCompiler.cpp]: Magic numbers!

File: loom/engine/loom2d/l2dQuad.cpp (hunk 3 above)
Comment (themuflon) [inline loom/engine/loom2d/l2dQuad.cpp]: It's from a merge

Comment (SmilyOrg) [inline loom/engine/loom2d/l2dQuad.cpp]: Oh? If that code is already in master I guess it's just github being weird :)

File: loom/script/compiler/lsTypeCompilerBase.cpp [hunk 11]
```diff
@@ -153,13 +178,663 @@ void TypeCompilerBase::generateIdentifierTypeConversion(
...
+}
+
+CompilationUnit *TypeCompilerBase::visit(CompilationUnit *cunit)
+{
+    return cunit;
+}
+
+Statement *TypeCompilerBase::visit(FunctionDeclaration *declaration)
+{
+    lmAssert(0, "FunctionDeclaration");
```
Comment (SmilyOrg) [inline loom/script/compiler/lsTypeCompilerBase.cpp]: Always fails?

File: loom/script/compiler/lsTypeCompilerBase.cpp [hunk 12]
```diff
@@ -153,13 +178,663 @@ void TypeCompilerBase::generateIdentifierTypeConversion(
...
+
+    return declaration;
+}
+
+
+Statement *TypeCompilerBase::visit(PropertyDeclaration *declaration)
+{
+    lmAssert(0, "PropertyDeclaration");
```
Comment (SmilyOrg) [inline loom/script/compiler/lsTypeCompilerBase.cpp]: Always fails?

File: loom/script/compiler/lsTypeCompilerBase.cpp [hunk 13]
```diff
@@ -153,13 +178,663 @@ void TypeCompilerBase::generateIdentifierTypeConversion(
...
+Statement *TypeCompilerBase::visit(ExpressionStatement *statement)
+{
+    statement->expression->visitExpression(this);
+    
+    return statement;
+}
+
+Statement *TypeCompilerBase::visit(LabelledStatement *statement)
+{
+    lmAssert(0, "LabelledStatement");
```
Comment (SmilyOrg) [inline loom/script/compiler/lsTypeCompilerBase.cpp]: Always fails?

File: loom/script/compiler/lsTypeCompilerBase.cpp [hunk 14]
```diff
@@ -153,13 +178,663 @@ void TypeCompilerBase::generateIdentifierTypeConversion(
...
+}
+
+Expression *TypeCompilerBase::visit(SuperExpression *expression)
+{
+    lmAssert(currentMethod, "Super call outside of method");
+
+    Type *type     = currentMethod->getDeclaringType();
+    Type *baseType = type->getBaseType();
+
+    //FIXME:  issue a warning
```
Comment (SmilyOrg) [inline loom/script/compiler/lsTypeCompilerBase.cpp]: Fix it? :)

File: loom/script/compiler/lsTypeCompilerBase.cpp [hunk 15]
```diff
@@ -153,13 +178,663 @@ void TypeCompilerBase::generateIdentifierTypeConversion(
...
+        return expression;
+    }
+
+    FuncState *fs = cs->fs;
+
+    if (currentMethod->isConstructor())
+    {
+        ConstructorInfo *base = baseType->getConstructor();
+
+        //FIXME: warn if no base constructor
```
Comment (SmilyOrg) [inline loom/script/compiler/lsTypeCompilerBase.cpp]: Also this.

File: loom/script/compiler/lsTypeCompilerBase.cpp [hunk 16]
```diff
@@ -153,13 +178,663 @@ void TypeCompilerBase::generateIdentifierTypeConversion(
...
+    {
+        utString name = currentMethod->getName();
+
+        if (expression->method)
+        {
+            name = expression->method->string;
+        }
+
+        MemberInfo *mi = baseType->findMember(name.c_str());
+        //FIXME: warn
```
Comment (SmilyOrg) [inline loom/script/compiler/lsTypeCompilerBase.cpp]: And this?

File: loom/script/compiler/lsTypeCompilerBase.cpp [hunk 17]
```diff
@@ -153,13 +178,663 @@ void TypeCompilerBase::generateIdentifierTypeConversion(
...
+    lmAssert(!declaration->classDecl, "local variable declaration belongs to a class");
+
+    ExpDesc v;
+    BC::singleVar(cs, &v, declaration->identifier->string.c_str());
+
+    Type       *dt     = declaration->type;
+    MethodInfo *method = NULL;
+    method = (MethodInfo *)dt->findMember("__op_assignment");
+
+    //TODO: assignment overload in default args?
```
Comment (SmilyOrg) [inline loom/script/compiler/lsTypeCompilerBase.cpp]: Needed before landing?

File: loom/script/native/lsNativeInterface.cpp [hunk 18]
```diff
@@ -409,6 +411,32 @@ void *lualoom_getnativepointer(lua_State *L, int index, bool replaceIndex, const
...
+            // uses 'key' (at index -2) and 'value' (at index -1)
+            printf("%x - %s *%x\n", ud, ud->m_nativeType->getCTypeName().c_str(), ud->getPointer());
+        }
+
+        // removes 'value'; keeps 'key' for next iteration
+        lua_pop(L, 1);
+    }
+    lua_pop(L, 1);
+}
+#endif
```
Comment (SmilyOrg) [inline loom/script/native/lsNativeInterface.cpp]: What's this? Dead code?

File: loom/script/native/lsNativeInterface.cpp [hunk 19]
```diff
@@ -418,6 +446,8 @@ void lualoom_downcastnativeinstance(lua_State *L, int instanceIdx, Type *fromTyp
     lmAssert(toType->isDerivedFrom(fromType), "lualoom_downcastnativeinstance - downcasting unrelated type %s to type %s", fromType->getFullName().c_str(), toType->getFullName().c_str());
     lmAssert(!toType->isInterface() && !fromType->isInterface(), "lualoom_downcastnativeinstance - downcasting interface type in cast type %s to type %s", fromType->getFullName().c_str(), toType->getFullName().c_str());
 
+    //lualoom_dumpmanagedtable(L, fromType, toType);
```
Comment (SmilyOrg) [inline loom/script/native/lsNativeInterface.cpp]: Debug code?

File: loom/script/native/lsNativeInterface.cpp [hunk 20]
```diff
@@ -485,6 +515,8 @@ void lualoom_downcastnativeinstance(lua_State *L, int instanceIdx, Type *fromTyp
     }
 
     lua_settop(L, top);
+
+    //lualoom_dumpmanagedtable(L, toType, fromType);
```
Comment (SmilyOrg) [inline loom/script/native/lsNativeInterface.cpp]: Debug code?

File: loom/script/runtime/lsClassRT.cpp [hunk 21]
```diff
@@ -41,47 +41,49 @@ void lsr_createinstance(lua_State *L, Type *type)
...
         lsr_getclasstable(L, t);
 
         // get the instance initializer function
         // this is not the constructor, which is a method
         lua_getfield(L, -1, "__ls_instanceinitializer");
 
-        lua_pushvalue(L, instanceIdx);
-
-        // call with new instance as "this" local
+        //LSLog(LSLogError, "   o initializer for %s", t->getFullName().c_str());
```
Comment (SmilyOrg) [inline loom/script/runtime/lsClassRT.cpp]: Debug code?

File: loom/script/runtime/lsClassRT.cpp [hunk 22]
```diff
@@ -127,19 +126,103 @@ void lualoom_newscriptinstance_internal(lua_State *L, Type *type)
     lua_setmetatable(L, instanceIdx);
 }
 
+#if 0
+    // useful debug stuff to be cleaned up
```
Comment (SmilyOrg) [inline loom/script/runtime/lsClassRT.cpp]: Now or later? :)

File: loom/script/runtime/lsClassRT.cpp [hunk 23]
```diff
@@ -127,19 +126,103 @@ void lualoom_newscriptinstance_internal(lua_State *L, Type *type)
...
 
+    // Note how many args we were called with.
     int nargs = lua_gettop(L) - 1;
 
     // index 1 on stack is class table
-
     Type *type = (Type *)lua_topointer(L, lua_upvalueindex(1));
 
-    bool profiling = LSProfiler::isEnabled();
+    //LSLog(LSLogError, "Creating instance: %s", type->getFullName().c_str());
```
Comment (SmilyOrg) [inline loom/script/runtime/lsClassRT.cpp]: Debug code?

File: loom/script/runtime/lsClassRT.cpp [hunk 24]
```diff
@@ -166,205 +250,167 @@ static int lsr_classcreateinstance(lua_State *L)
         lua_rawseti(L, instanceIdx, LSINDEXGCTRACKER);
     }
 
+    // If it has a native base type, instantiate that via C++ first.
+    Type *nt = type->getNativeBaseType();
+    if(nt)
+    {
+        //LSLog(LSLogError, "   o creating native: %s", nt->getFullName().c_str());
```
Comment (SmilyOrg) [inline loom/script/runtime/lsClassRT.cpp]: Also here.

File: loom/script/runtime/lsClassRT.cpp [hunk 25]
```diff
@@ -166,205 +250,167 @@ static int lsr_classcreateinstance(lua_State *L)
...
         }
 
-        t = t->getBaseType();
-    }
-
-    // instance intializer chain
-    lualoom_callscriptinstanceinitializerchain_internal(L, type, instanceIdx, NULL);
-
-    // call constructor chain
+        //LSLog(LSLogError, "   o native ctor %s", type->getFullName().c_str());
```
Comment (SmilyOrg) [inline loom/script/runtime/lsClassRT.cpp]: And here.

File: loom/script/runtime/lsClassRT.cpp [hunk 26]
```diff
@@ -166,205 +250,167 @@ static int lsr_classcreateinstance(lua_State *L)
...
+        {
+            LSLuaState *ls = LSLuaState::getLuaState(L);
+            ls->triggerRuntimeError("Error creating instance for native class %s\n\n"
+                                    "Please make sure you have a constructor defined in your native bindings\n"
+                                    "For example: .addConstructor <void (*)(void) >()",
+                                    type->getFullName().c_str());
+        }
 
-    int top = lua_gettop(L);
+        //LSLog(LSLogError, "            - done");
```
Comment (SmilyOrg) [inline loom/script/runtime/lsClassRT.cpp]: And here!

File: loom/script/runtime/lsClassRT.cpp [hunk 27]
```diff
@@ -166,205 +250,167 @@ static int lsr_classcreateinstance(lua_State *L)
...
 
-    t = types.pop();
-    while (t)
-    {
-        ConstructorInfo *cinfo = t->getConstructor();
+        lua_rawseti(L, instanceIdx, LSINDEXNATIVE);
 
-        bool cskip = t->isNative() && (cinfo && cinfo->isNative());
+        // Sanity check that this is a known native type.
+        assert(NativeInterface::getNativeType(nt));
```
Comment (SmilyOrg) [inline loom/script/runtime/lsClassRT.cpp]: lmAssert?

File: loom/script/runtime/lsClassRT.cpp [hunk 28]
```diff
@@ -1082,15 +1127,19 @@ void lsr_classinitialize(lua_State *L, Type *type)
...
         {
-            LSError("Error loading bytecode for %s:%s",
+            // We skip initializers that do nothing to save ops, so missing byte
+            // code isn't a problem.
+/*            LSWarning("Error loading bytecode for %s:%s",
                     type->getFullName().c_str(),
                     i == 0 ?
                     "__ls_staticinitializer" :
-                    "__ls_instanceinitializer");
+                    "__ls_instanceinitializer"); */
```
Comment (SmilyOrg) [inline loom/script/runtime/lsClassRT.cpp]: Dead code?

File: sdk/src/test/tests/TestManagedNativeClass.ls [hunk 29]
```diff
@@ -85,69 +83,137 @@ final native class MyChildManagedNativeClass extends MyManagedNativeClass {
...
 {
     function retrieveChild(owner:MyManagedNativeClass) {
         
-        // get the child, and make sure the script value is still valid
-        // magic! 
+        // get the child, and make sure the script value is still valid magic! 
         var child = owner.getChild();
-        log(child.scriptString);
-        log((child as MyChildManagedNativeClass).scriptStringChild);
+        Assert.equal(child.scriptString, "Hello, ");
```
Comment (SmilyOrg) [inline sdk/src/test/tests/TestManagedNativeClass.ls]: `Assert.compare` might be more appropriate for this one and the ones below. Functionally the same, but semantically different as `compare` is used to compare the runtime result value with a predefined expected value (and the error message on failure reflects that) whereas equals is just a generic equality check.

File: sdk/src/test/tests/TestManagedNativeClass.ls [hunk 30]
```diff
@@ -85,69 +83,137 @@ final native class MyChildManagedNativeClass extends MyManagedNativeClass {
     {
         super.intProperty = value + 2;
     }
-
-    
 }
 
-class TestManagedNativeClass extends LegacyTest
+class TestManagedNativeClass
```
Comment (SmilyOrg) [inline sdk/src/test/tests/TestManagedNativeClass.ls]: The new tests usually have `Test` at the end of the name, but it's not a big deal.

File: loom/script/native/core/system/lmConsole.cpp [hunk 31]
```diff
@@ -36,8 +36,9 @@ int Console::print(lua_State *L)
     lua_rawgeti(L, 1, LSINDEXVECTOR);
     int vidx = lua_gettop(L);
 
-    // the value to print (concating all objects passed)
-    utString toprint;
+    // the value to print (concatenating all objects passed)
+    char outputBuffer[2048];
```
Comment (SmilyOrg) [inline loom/script/native/core/system/lmConsole.cpp]: Maybe we should just transition to using the stack buffer if possible (total length being < 2048 or whatever) and allocating if there's too much to print. Perhaps for a different PR.

**Comment by SmilyOrg** [comment]
I went through the code and noted some nits, except for the graphics stuff that apparently should already be in master. I also haven't tested the functionality yet and I can't speak for lua-specific code as it's somewhat opaque :)

Also it failed build, so that's something to look at.

File: docs/examples/ScriptBenchmark/src/BenchmarkAllocation.ls (hunk 1 above)
Comment (bengarney) [inline docs/examples/ScriptBenchmark/src/BenchmarkAllocation.ls]: Updated.

File: loom/engine/loom2d/l2dQuad.cpp (hunk 3 above)
Comment (bengarney) [inline loom/engine/loom2d/l2dQuad.cpp]: Yeah I merged and that's where it came from. It should not be present.

File: loom/script/compiler/lsJitTypeCompiler.cpp (hunk 4 above)
Comment (bengarney) [inline loom/script/compiler/lsJitTypeCompiler.cpp]: Mystery code leftover. Not prepared to definitively investigate now.

File: loom/script/compiler/lsJitTypeCompiler.cpp (hunk 5 above)
Comment (bengarney) [inline loom/script/compiler/lsJitTypeCompiler.cpp]: No, useful debugging. I've replaced it with a macro.

File: loom/script/compiler/lsJitTypeCompiler.cpp (hunk 6 above)
Comment (bengarney) [inline loom/script/compiler/lsJitTypeCompiler.cpp]: Done.

File: loom/script/compiler/lsJitTypeCompiler.cpp (hunk 7 above)
Comment (bengarney) [inline loom/script/compiler/lsJitTypeCompiler.cpp]: JIT and classic VMs have different names (dumb).

File: loom/script/compiler/lsJitTypeCompiler.cpp (hunk 8 above)
Comment (bengarney) [inline loom/script/compiler/lsJitTypeCompiler.cpp]: Potentially something to investigate someday.

File: loom/script/compiler/lsJitTypeCompiler.cpp (hunk 10 above)
Comment (bengarney) [inline loom/script/compiler/lsJitTypeCompiler.cpp]: Correlates to buffer above, but shorter than having an arbitrary #define and simpler than a sizeof macro.

File: loom/script/compiler/lsTypeCompilerBase.cpp (hunk 11 above)
Comment (bengarney) [inline loom/script/compiler/lsTypeCompilerBase.cpp]: Meant not to be called. I'm just mimicking the existing code here.

File: loom/script/compiler/lsTypeCompilerBase.cpp (hunk 14 above)
Comment (bengarney) [inline loom/script/compiler/lsTypeCompilerBase.cpp]: I didn't introduce this so not going to fix it in this PR.

File: loom/script/compiler/lsTypeCompilerBase.cpp (hunk 15 above)
Comment (bengarney) [inline loom/script/compiler/lsTypeCompilerBase.cpp]: Ditto.

File: loom/script/compiler/lsTypeCompilerBase.cpp (hunk 16 above)
Comment (bengarney) [inline loom/script/compiler/lsTypeCompilerBase.cpp]: Ditto.

File: loom/script/compiler/lsTypeCompilerBase.cpp (hunk 17 above)
Comment (bengarney) [inline loom/script/compiler/lsTypeCompilerBase.cpp]: Nope, leftover.

File: loom/script/native/lsNativeInterface.cpp (hunk 18 above)
Comment (bengarney) [inline loom/script/native/lsNativeInterface.cpp]: Useful debug code. I've cleaned this up to be better documented and also clarify why the calls to it are commented out but kept.

File: loom/script/runtime/lsClassRT.cpp (hunk 21 above)
Comment (bengarney) [inline loom/script/runtime/lsClassRT.cpp]: Made this macro-fied to remain useful.

File: loom/script/runtime/lsClassRT.cpp (hunk 23 above)
Comment (bengarney) [inline loom/script/runtime/lsClassRT.cpp]: Macrofied.

File: loom/script/runtime/lsClassRT.cpp (hunk 24 above)
Comment (bengarney) [inline loom/script/runtime/lsClassRT.cpp]: Macrofied.

File: loom/script/runtime/lsClassRT.cpp (hunk 25 above)
Comment (bengarney) [inline loom/script/runtime/lsClassRT.cpp]: Macrofied.

File: loom/script/runtime/lsClassRT.cpp (hunk 26 above)
Comment (bengarney) [inline loom/script/runtime/lsClassRT.cpp]: Macrofied.

File: loom/script/runtime/lsClassRT.cpp (hunk 27 above)
Comment (bengarney) [inline loom/script/runtime/lsClassRT.cpp]: It's leftover code so I am leaving it as is.

File: loom/script/runtime/lsClassRT.cpp (hunk 28 above)
Comment (bengarney) [inline loom/script/runtime/lsClassRT.cpp]: Macrofied.

File: sdk/src/test/tests/TestManagedNativeClass.ls (hunk 29 above)
Comment (bengarney) [inline sdk/src/test/tests/TestManagedNativeClass.ls]: OK - fixed!

File: sdk/src/test/tests/TestManagedNativeClass.ls (hunk 30 above)
Comment (bengarney) [inline sdk/src/test/tests/TestManagedNativeClass.ls]: Fixed!

File: loom/script/native/core/system/lmConsole.cpp (hunk 31 above)
Comment (bengarney) [inline loom/script/native/core/system/lmConsole.cpp]: Can you please add a trello card for this? I know alloca is sometimes finicky cross platform so it's worth exploring but out of scope for this PR.

**Comment by bengarney** [comment]
Responded to nits. I will debug build when back on build box network.

**Comment by bengarney** [comment]
Does it build for you on Windows?

**Comment by bengarney** [comment]
Windows passes, landing my own PR (scary)!

**Pull request merged**